rapids-4-spark-tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/QualificationInfoUtils.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.profiling
+
+import java.io.File
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.rapids.TestUtils
+import org.apache.spark.sql.{Dataset, SparkSession}
+import org.apache.spark.sql.functions.{col, udf}
+
+// class used for testing
+case class RapidsFriends(name: String, friend: String, age: Int)
+
+/**
+ * Utilities to generate event logs used for qualification testing.
+ */
+object QualificationInfoUtils extends Logging {
+
+  def randomStringFromCharList(length: Int, chars: Seq[Char]): String = {
+    val sb = new StringBuilder
+    for (i <- 1 to length) {
+      val randomNum = util.Random.nextInt(chars.length)
+      sb.append(chars(randomNum))
+    }
+    sb.toString
+  }
+
+  def randomAlpha(length: Int): String = {
+    val chars = ('a' to 'z') ++ ('A' to 'Z')
+    randomStringFromCharList(length, chars)
+  }
+
+  def randomString(length: Int) = {
+    val r = new scala.util.Random
+    val sb = new StringBuilder
+    for (i <- 1 to length) {
+      sb.append(r.nextPrintableChar)
+    }
+    sb.toString
+  }
+
+  val randForInt = new scala.util.Random(11)
+  def randomInt(): Int = {
+    randForInt.nextInt(100)
+  }
+
+  def generateFriendsDataset(spark: SparkSession): Dataset[RapidsFriends] = {
+    import spark.implicits._
+    val df = spark.sparkContext.parallelize(
+      Seq.fill(1000){(randomAlpha(10), randomAlpha(5), randomInt)})
+      .toDF("name", "friend", "age")
+    df.as[RapidsFriends]
+  }
+
+  // dataset operations in plan show up as Lambda
+  def genDatasetEventLog(spark: SparkSession, size: Int = 1000) = {
+    import spark.implicits._
+    TestUtils.withTempPath { jsonOutFile =>
+      val ds = generateFriendsDataset(spark)
+      val dsAge = ds.filter(d => d.age > 25).map(d => (d.friend, d.age))
+      dsAge.write.json(jsonOutFile.getCanonicalPath)
+    }
+  }
+
+  def parseAge = (age: Int) => {
+    val majorAge = Seq("21", "25", "55", "18")
+    if (majorAge.contains(age)) {
+      "MILESTONE"
+    } else {
+    else {
```

Comment:
Sorry, extra `else` from here and `cleanCountry`  should be deleted. I had made wrong substitution while suggesting.

Reply:
fixing